dask/dataframe/groupby.py

```diff
@@ -305,11 +305,15 @@ def _var_chunk(df, *index):
 
     n = g[x.columns].count().rename(columns=lambda c: (c, "-count"))
 
-    df[cols] = df[cols] ** 2
+    try:
+        for col in set(cols).difference(set(index)):
+            df[col] = df[col] ** 2
```

Comment:
Why this change?  What was going on?

Reply:
The issue is that the `by` key is being squared. For large `int64` values in particular this will produce either clamping from `cudf` or overflow with `pandas`. The failure is more obvious with `cudf` because all values greater than `np.iinfo(np.int64).max**0.5` clamp to max. With pandas, all multiples of `np.iinfo(np.in64).max**0.5` wrap back around to `0`. There are probably other surjections in pandas but I can't map them all.

Reply:
I did it in a `try:` because `index` could be a `cudf` or `pandas` `Series` (and maybe something else? `level`?). I'm not convinced that's the right option. Now that I figured out the Pandas reproducer I can get back to writing a better fix.

Reply:
@thomcom do you have a better fix in mind currently, or is that more of a TODO in your mind?

Reply:
I'm essentially happy with it as is, but I'm giving reviewers the chance to disapprove of my `try/catch` method. :) It isn't really necessary, but try/catch can always hide other errors.

Reply:
If we succeed at squaring some of the columns, but then fail on one, it looks like we end up squaring some of them twice?

Reply:
Also, if we run into errors in the try block then why don't we get those same errors in the except block?  What is different there.

Reply:
Do we actually get an exception with the overflow/clamping?  Or do we just get a wrong value?

Reply:
I think the bug is on L301. What if you change

```
cols = df._get_numeric_data().columns
```

to

```python
cols = x.columns
```

after L304? Then things may work out.

Reply:
The `try` actually handles the `set(cols).difference(set(index)):` call, as `index` could be a column or a list of columns, or it could be a `Series` (or possibly something else I am not anticipating, hence the `try`). The objective here is just to square the columns that aren't listed in `by`. If that fails, then assume that `index` is not a list of `by` columns, and square everything on L312.

The exception does not occur during the overflow/clamping, but due to the surjection causes L315 to produce a difference set of key rows for `x2` than it did for `x`. The exception is then thrown on L312 because `x.index` and `x2.index` have a different number of rows.